### DISCUSSION TITLE
MIDI out on Native Instument Traktor controllers

### DISCUSSION THREAD

**Discussion by richardbenson** [cpyarger/obs-midi#96] [Q&A] [answered]
Hi there, I assume I am doing something wrong but I cannot seem to get the bidirectional working. I am a little confused about how exactly it should be configured; I have scence changes and record/buffer toggles set up and working, all on control changes but there never seems to be any incoming midi.  I have enabled bidirectional in the devices panel.

Using Native Instruments Traktor X1 in MIDI mode, commit b67d702.

(Note the other devices are not enabled)

![image](https://user-images.githubusercontent.com/98186/110103799-41f7dd80-7d9e-11eb-8a15-71aaf02329c5.png)

![image](https://user-images.githubusercontent.com/98186/110103821-49b78200-7d9e-11eb-91a6-c28e3e3292bf.png)

**Comment by cpyarger** [comment, marked as answer]
It currently sends signals back on the same channel and note or control they come in on, I am looking into a good way to be able to remap outputs, but for the moment I have it set up this way to give some feedback. 

Also, Its made with the assumption that notes or controls are used as was originally designed, notes for buttons, control changes for faders or encoders.

**Reply by richardbenson** [reply]
Interesting, all the NI Traktor stuff defaults the mappings on every button, pot and fader to CC, I hadn't thought to try notes and they work perfectly!

I did start by saying I was probably doing something wrong ;)

Thanks!

**Reply by cpyarger** [reply]
There is also a first time use bug, at the moment that prevents feedback until you restart obs the first time you enable a device
